src/components/Video.astro

```diff
@@ -0,0 +1,32 @@
+---
+import { getLanguageFromURL } from '../util';
+import languages from '../i18n/languages';
+
+export interface Props {
+	src: string;
+	type: string;
+}
+
+const { src, type } = Astro.props as Props;
+const url = new URL(Astro.request.url);
+const lang = getLanguageFromURL(url.pathname);
+const filename = src.substring(src.lastIndexOf('/') + 1, src.lastIndexOf('.'));
+
+const allCaptions = import.meta.glob("../../public/captions/**/*.vtt", { as:'raw' });
+const fileCaptions = Object.keys(allCaptions).map((caption) => {
+	const captionPath = caption.substring(caption.indexOf('captions') - 1)
+	const captionLang = captionPath.substring(10, captionPath.indexOf(filename) - 1);
+	return {
+		path: captionPath,
+		lang: captionLang
+	}
+}).filter(({path}) => path.includes(filename))
```

Comment:
Still in the realm of slightly subjective, but it took me a minute to read through this and understand, so I’ve tried pulling the directory into a named variable so we can use its length directly. I also renamed the other variables to simplify the returned object. What do you think?

```suggestion
const fileCaptions = Object.keys(allCaptions)
	.map((caption) => {
		const dir = '/captions/';
		const path = caption.substring(caption.indexOf(dir));
		const lang = path.substring(dir.length, path.indexOf(filename) - 1);
		return { path, lang };
	})
	.filter(({ path }) => path.includes(filename));
```